turbopack/crates/turbopack-ecmascript-runtime/js/src/browser/runtime/base/dev-base.ts

```diff
@@ -62,6 +62,38 @@ class UpdateApplyError extends Error {
   }
 }
 
+/**
```

Comment:
nice! i just noticed that a lot of these `hmr` datastructures were in the wrong file.